release_docs/RELEASE_PROCESS.md

```diff
@@ -0,0 +1,109 @@
+
+## Description (General):
+At regularly scheduled intervals throughout each year, The HDF Group releases a new version of each of its maintained / core software products and corresponding documentation set. This is referred to as a maintenance release. The main purpose of a maintenance release is to address defects reported against previous releases of the software, to port the software to new operating environments, and to improve maintainability by way of refactoring code and improving documentation. A maintenance release may also include new functionality in any of the product's components.
+
+Depending on the software being released, there may be specific guidelines that are followed or steps that are performed in addition to the 'standard' maintenance release procedure - in such a case, product-specific tabs are provided within the page to view information specific to only that product.
+
+## Description (HDF5):
+Twice each year, The HDF Group releases a new version of the HDF5 C, C++, and Fortran libraries and command line utilities and associated documentation set. This is referred to as a maintenance release of HDF5. The main purpose of a maintenance release is to address defects reported against previous releases of the software, to port HDF5 to new operating environments, and to improve maintainability by way of refactoring code and improving documentation. A maintenance release may also include new functionality in the C, C++, and Fortran libraries and/or command line utilities if requested and/or funded by the user community and/or The HDF Group's customers. The new features included in a maintenance release expand the feature set only and do not change the HDF5 file format, with the exception of addressing critical defects that corrupt data.
+
+Maintenance releases are always backward compatible with regards to the HDF5 file format, meaning that:
+- New libraries and command line utilities can access HDF5 files created by the previous versions of the libraries.
+
+Maintenance releases are always forward compatible with regards to the HDF5 file format, meaning that:
+- HDF5 libraries and command line utilities can access files created by future maintenance versions of the library.
+Note that maintenance releases are NOT guaranteed to be interface-compatible, meaning that, on occasion, application source code will need updated and re-compiled against a new maintenance release when the interface changes. Interface changes are only made when absolutely necessary as deemed by the HDF5 product manager(s), and interface compatibility reports are published with each release to inform customers and users of any incompatibilities in the interface.
+
+For more information on the HDF5 versioning and backward and forward compatibility issues, see the [HDF5 FAQ](http://www.hdfgroup.org/HDF5-FAQ.html) on the public website.
```

Comment:
```suggestion
For more information on the HDF5 versioning and backward and forward compatibility issues, see the [HDF5 FAQ](https://docs.hdfgroup.org/hdf5/v1_14/api-compat-macros.html) on the public website.
```

Reply:
These two suggestions, when committed, should allow the link checks to pass.